mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -250,114 +240,142 @@ class KMeans private (
         }
       }
     }
+
     val initTimeInSeconds = (System.nanoTime() - initStartTime) / 1e9
     logInfo(s"Initialization with $initializationMode took " + "%.3f".format(initTimeInSeconds) +
       " seconds.")
 
-    val active = Array.fill(numRuns)(true)
-    val costs = Array.fill(numRuns)(0.0)
-
-    var activeRuns = new ArrayBuffer[Int] ++ (0 until numRuns)
+    var costs = 0.0
     var iteration = 0
-
     val iterationStartTime = System.nanoTime()
+    val isSparse = data.take(1)(0).vector.isInstanceOf[SparseVector]
 
-    // Execute iterations of Lloyd's algorithm until all runs have converged
-    while (iteration < maxIterations && !activeRuns.isEmpty) {
+    // Execute Lloyd's algorithm until converged or reached the max number of iterations
+    while (iteration < maxIterations) {
```

Comment:
Won't the algorithm always run until `maxIterations` with this change? The convergence check inside the loop is unused.